homeassistant/helpers/event.py

```diff
@@ -1352,39 +1364,12 @@ def pattern_time_change_listener() -> None:
         calculate_next(now + timedelta(seconds=1))
 
         cancel_callback = hass.loop.call_at(
-            -time.time()
-            + hass.loop.time()
-            + next_time.timestamp()
-            + MAX_TIME_TRACKING_ERROR,
+            _datetime_to_loop_time(hass, next_time),
             pattern_time_change_listener,
         )
 
-    # We always get time.time() first to avoid time.time()
-    # ticking forward after fetching hass.loop.time()
-    # and callback being scheduled a few microseconds early.
-    #
-    # Since we loose additional time calling `hass.loop.time()`
-    # we add MAX_TIME_TRACKING_ERROR to ensure
-    # we always schedule the call within the time window between
-    # second and the next second.
-    #
-    # For example:
```

Comment:
Let's keep the example in the code.

Reply:
Examples are nice but I think the example is wrong. I will set to draft while I think about a different phrasing.